snooty/rstparser.py

```diff
@@ -696,10 +696,17 @@ class Domain:
 
 @dataclass
 class Registry:
+    default_domain: Optional[str] = None
     domains: DefaultDict[str, Domain] = field(
         default_factory=lambda: defaultdict(Domain)
     )
 
+    def get_domain_resolution_sequence(self) -> List[str]:
+        # If default domain has been specified, prepend it to the list of domains
+        if self.default_domain:
+            return [self.default_domain] + DOMAIN_RESOLUTION_SEQUENCE
```

Comment:
This will involve constructing *two* new list instances upon every directive/role lookup. This should probably be a precomputed attribute of the Registry.

Bonus points if directive/role resolution is factored into methods of this class, maybe?

Reply:
Agree with this! The only reason I didn't do this is because I had some trouble figuring out how to do that within the confines of `@dataclass`. Maybe the most effective way to fix this would be for you to pull this branch down and make that edit, given that we are out tomorrow. Let me know!

Reply:
While it is possible to have a dataclass with a constructor, this doesn't actually need to be a dataclass. Remove the `@dataclass` decorator, and just have a normal constructor instead of the field declarations.

Reply:
I tried getting into that briefly yesterday and ran in to trouble. I will leave as-is for now and mark with an `XXX` comment!